src/extensive_list_of_resrouce.md

```diff
@@ -0,0 +1,247 @@
+- [A Trip Through The Graphics Pipeline](https://alaingalvan.gitbook.io/a-trip-through-the-graphics-pipeline) [article]
+- [Introduction to Compute Shaders](https://anteru.net/blog/2018/intro-to-compute-shaders/) [article]
+- [GPU Architectures](https://drive.google.com/file/d/12ahbqGXNfY3V-1Gj5cvne2AH4BFWZHGD/view) [slides]
+- [Introduction to Computer Graphics](https://math.hws.edu/graphicsbook/index.html) [book]
+- [Build Your Own 3D Renderer](https://avikdas.com/build-your-own-raytracer/) [article]
+
+## OpenGL
+- [Learn OpenGL](https://learnopengl.com/) [book]
+- [OpenGL Step By Step](https://ogldev.org/) [video]
+- [OpenGL Tutorials](https://www.opengl-tutorial.org/) [book]
+- [OpenGL Introduction](https://open.gl/introduction) [article]
+- [Anton's OpenGL 4 Tutorials](https://antongerdelan.net/opengl/) [book] [article]
+- [OpenGL Is Not Right-Handed](https://www.gingerbill.org/article/2024/11/10/opengl-is-not-right-handed/) [article]
+- [Best Practices for Modern OpenGL](https://juandiegomontoya.github.io/modern_opengl.html) [article]
+- [A Guide to Modern OpenGL Functions](https://github.com/fendevel/Guide-to-Modern-OpenGL-Functions) [article]
+- [OpenGl Tutorials](https://www.youtube.com/playlist?list=PLPaoO-vpZnumdcb4tZc4x5Q-v7CkrQ6M-) [video]
+- [Docs GL - OpenGL Documentation](https://docs.gl/) [documentation]
+- [How to write a renderer for modern graphics APIs](https://blog.mecheye.net/2023/09/how-to-write-a-renderer-for-modern-apis/) [article]
+- [MBSoftworks OpenGL Tutorials](https://www.mbsoftworks.sk/tutorials/) [article]
+- [An Intro to Modern OpenGL](https://duriansoftware.com/joe/an-intro-to-modern-opengl.-table-of-contents) [book]
+- [OpenGL Hardware Database](https://opengl.gpuinfo.org/) [other]
+- [A Hitchhiker's Guide to OpenGL](https://github.com/bartvbl/A-Hitchhikers-Guide-to-OpenGL) [book]
+- [How to make OpenGL usage Vulkan like](https://developer.nvidia.com/opengl-vulkan)
+
+## Shaders
+- [3D Game Shaders For Beginners](https://lettier.github.io/3d-game-shaders-for-beginners/index.html)
+- [The Book of Shaders](https://thebookofshaders.com/)
+- [Shaders Monthly](https://www.youtube.com/playlist?list=PL8vNj3osX2PzZ-cNSqhA8G6C1-Li5-Ck8)
+- [Shadertoy](https://www.shadertoy.com/)
+- [GLSL 2D Tutorials](https://www.shadertoy.com/view/Md23DV)
+- [Shader School](https://github.com/stackgl/shader-school)
+- [OpenGL 4 Shaders](https://antongerdelan.net/opengl/shaders.html)
+- [Introduction to Shader Art Coding](https://www.youtube.com/watch?v=f4s1h2YETNY)
+- [A Beginner's Guide to Coding Graphic Shaders](https://code.tutsplus.com/series/a-beginners-guide-to-coding-graphics-shaders--cms-834) [article]
+
+## Tooling (Debuggers)
+- [RenderDoc](https://renderdoc.org/)
+- [NVIDIA Nsight Graphics](https://developer.nvidia.com/nsight-graphics)
+- [PIX](https://devblogs.microsoft.com/pix/introduction/)
+- [Rad Debugger](https://github.com/EpicGamesExt/raddebugger)
```

Comment:
"Rad" should be 'RAD'.
This should probably get a quick spelling and capitalization check.